File: lab 3/SingleEvent.cpp
#include <iostream>
#include "SingleEvent.h"
using namespace std;

SingleEvent::SingleEvent() :Event()
{
    for (int i : date) {
        i = 0;
    }
}

SingleEvent::SingleEvent(const string& name, double duration, int dd, int mm, int yy): Event(name, duration)
{
    date[0] = dd; date[1] = mm; date[2] = yy;

}

void SingleEvent::displayEvent() const
{
    cout << "Single Event: " << name 
        << ", " << date[0] << "." << date[1] << "." << date[2] << ", Duration: " << duration << endl;
}

SingleEvent& SingleEvent::inputEventDetails()
{
    cout << "Enter information about event: name, duration, date(day, month, year)" << endl;
    cin >> name >> duration >> date[0] >> date[1] >> date[2];
    return *this;
}

const int* SingleEvent::getDate() const {
    return date;
}

File: python/SingleEvent.py
from Event import Event

class SingleEvent(Event):
    def __init__(self, line, dur, dd, mm, yy):
        super().__init__(line, dur)
        self._date=[dd, mm, yy]
    
    def get_date(self):
        return self._date
    
    def display_event(self):
        print(f"{self._name}: {self._date[0]}.{self._date[1]}.{self._date[2]}, Duration: {self._duration}")
        
    def input_event_details(self):
        print(f"Enter information about event:\n")
        self._name=input("name: ")
        self._duration=float(input("duration: "))
        self._date[0]=int(input("day: "))
        self._date[1]=int(input("month: "))
        self._date[2]=int(input("year: "))
